lua/flatten/guest.lua

```diff
@@ -22,7 +22,7 @@ function M.maybe_block(block)
   end
   vim.fn.chanclose(host)
   while true do
-    vim.cmd.sleep(1)
+    vim.cmd.sleep(10)
```

Comment:
I think maybe this should just be changed to `vim.wait` - I will look into this now. I wrote the original code for this quite a while ago when I was a lot less familiar with Neovim's internals and APIs.

Reply:
There's a big problem with `vim.wait` - it is async, meaning guest stays fully responsive which is what we do *not* want because a user can accidentally mess up guest. `sleep` on the other hand fully blocks guest neovim instance.

Reply:
It is blocking the function where it is called but it is not blocking the UI.

Try this:
1. Open terminal: `:term`
2. Add something to index
3. `git commit`
4. Return to the terminal buffer in host
5. `i` once, you will enter terminal mode in host <- this is expected
6. `i` second time, you will enter insert mode in guest <- this should not happen but it does because `vim.wait` only blocks `maybe_block`

That's why I believe it is better to use `sleep`.